test/server_sent_event_stage_test.exs

```diff
@@ -194,10 +194,60 @@ defmodule ServerSentEventStageTest do
       GenStage.stop(pid)
     end
 
-    defp start_producer(bypass) do
+    test "applies idle timeout", %{bypass: bypass} do
```

Comment:
question: would this be simpler (and still test the same behavior) with something like this (high-level):
```elixir
Bypass.stub(bypass, "GET", "/", fn conn -> 
  Process.sleep(idle_timeout - 10)
  # send comment
  Process.sleep(idle_timeout - 10)
  # send data chunk
  Process.sleep(:infinity)
end)
# assert received data chunk
# assert received timeout
```

Mostly, by using `stub/4` instead of `expect/4`?

Reply:
Using `stub` doesn't solve the problem, because Bypass will still monitor the process and fail when the request is cancelled. There is a [correct solution](https://hexdocs.pm/bypass/Bypass.html#pass/1), though.

There's no direct way to test that the timeout was received, though; what we really want to test is that the connection is retried, and the most explicit way to do that is to stick with counting the requests.